halo2_proofs/src/dev.rs

```diff
@@ -327,6 +341,10 @@ impl<F: Field + Group> Assignment<F> for MockProver<F> {
     }
 
     fn exit_region(&mut self) {
+        if !self.in_phase(FirstPhase) {
+            return;
+        }
```

Comment:
Same question, qhy do we need to be in a phase that isn't `FirstPhase`

Reply:
ditto